src/features/applications/index.ts

```diff
@@ -1,13 +1,15 @@
+import { errors } from '@balena/pinejs';
+
 export const getApplicationSlug = (
 	orgHandle: string,
 	appName: string,
 ): string => {
 	if (orgHandle == null) {
-		throw new Error('Missing organization handle');
+		throw new errors.BadRequestError('Missing organization handle');
```

Comment:
These are obviously behavior changes, but I do agree that it is for the better and don't expect anyone to be depending on this returning a 404.